### ISSUE TITLE
virtio-gpu-gl-device doesn't boot, and virtio-gpu-gl-pci fails after guest reaches gpu

### ISSUE THREAD

**Issue by kode54** [utmapp/UTM#3084]
**Describe the issue**
The newly added GL support in 2.2.2 beta is not working properly. `virtio-gpu-gl-device` gets stuck at a black screen before even reaching firmware. `virtio-gpu-gl-pci` runs really slowly, as if each individual GL draw operation is gated by a vsync or flip operation, then eventually errors out when the guest reaches the GPU state. `virtio-ramfb` seems to run as if its "accelerated" GL is actually a softgpu.

**Configuration**
* UTM Version: 2.2.2
* OS Version: macOS Big Sur 11.5.2
* Intel or Apple Silicon? Apple Silicon

**Crash log**
Ah, there is a crash:
[QEMULauncher_2021-09-03-184719_Christophers-Mac-mini.crash.zip](https://github.com/utmapp/UTM/files/7112319/QEMULauncher_2021-09-03-184719_Christophers-Mac-mini.crash.zip)
[QEMULauncher_2021-09-05-134849_Christophers-Mac-mini.crash.zip](https://github.com/utmapp/UTM/files/7112338/QEMULauncher_2021-09-05-134849_Christophers-Mac-mini.crash.zip)

**Debug log**
For all issues, _including_ crashes, you should attach a debug log. Open UTM, and open the settings for the VM you wish to launch. Near the top of the `QEMU` page is `Debug Log`. Turn it on and save the VM. After you experience the issue, open the VM settings again and select `Export Log...` and attach it here.

**Upload VM**
Here is the config file for the VM:
[config.plist.zip](https://github.com/utmapp/UTM/files/7112343/config.plist.zip)

**Comment by osy** [comment]
@kode54 What are you trying to run? If possible can you upload the .utm?

**Comment by kode54** [comment]
Just trying to run Ubuntu Server 21.04 that has been configured to enable ubuntu-desktop. ARM version, on an M1 Mac mini. The problem with the cookie error doesn't happen very often now, but the performance of the virgl is atrocious, getting just over 106fps in non-vsync-constrained glxgears.

**Comment by osy** [comment]
What do you expect the performance to be? What do you get with QEMU?

**Comment by kode54** [comment]
I guess if I want Linux to run with gaming level performance on ARM hardware, I should buy a Linux specific ARM machine.

**Comment by phirestalker** [comment]
I could not reproduce a crash, but I do see the slow drawing of the console before graphics mode starts. When I use that display driver it has buttery smooth graphics (I only played a music video, no gaming). Here are some details about my setup. Maybe we can track down differences.

UTM: Version 2.2.4 (36)
OS: 11.5.2 (20G95)
Apple Silicon

My guest is Debian Buster installed on UTM (not converted from another virtualization program)
Now some supporting documentation

Debug log:
[debug.log](https://github.com/utmapp/UTM/files/7150609/debug.log)

UTM vm config
[config.plist.txt](https://github.com/utmapp/UTM/files/7150611/config.plist.txt)

A video clip of the slow drawing on the console
https://user-images.githubusercontent.com/7477095/133010506-f0c90a03-a57f-4f1c-ba63-d553f9939450.mp4

I doubt this is a UTM issue though.

**Comment by kode54** [comment]
Nope, I've gathered that virgl is not really all that performant. I've found it to actually be less performant at anything GPU heavy than llvmpipe.

**Comment by djlenadanger** [comment]
**I changed my Display to virtio-ramfb and that fixed it.** 

[v 4.1 Updates in the Note Section](https://docs.getutm.app/updates/v4.1/)

Newly created Linux VMs will now use virtio-gpu-gl-pci by default. It is recommended that you change the [display card](https://docs.getutm.app/settings-qemu/devices/display/#emulated-display-card) of existing QEMU backend Linux VMs to this card in order to take advantage of the improved renderer backend. This should result in improved performance in GUI rendering. Note that some applications (particularly 3D applications) may lock up or crash UTM and if you are experiencing issues, you can go back to virtio-ramfb or virtio-vga.

<img width="1024" alt="Screen Shot 2023-01-11 at 2 15 01 PM" src="https://user-images.githubusercontent.com/122492295/211930931-d781a0ed-9b95-4f1a-8342-a6baebbaba0d.png">
